### PR TITLE
[mqtt.ruuvigateway] Re-enable itests after core change

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Builds for #15010 somehow do not complete. This is a try to do it separately.

Re-enable itests for mqtt.ruuvigateway.

After core change ```CoreItemFactory``` ctor requires a ```UnitProvider```.
From itests the class ```TestUnitProvider``` is not visible, so I tried a mockup.

Jenkins results need to be reviewed.

### CODE REVIEW

**Review by wborn — COMMENTED** [review]
Try resolving the runbundles again:

https://github.com/wborn/openhab-addons#integration-tests

**Comment by holgerfriedrich** [comment]
Then let's wait for skinah/openhab-addons#4 and #14878....

**Comment by jlaur** [comment]
@holgerfriedrich, @lolodomo just FYI, it seems that #14970 caused #15019 to fail?

**Comment by jlaur** [comment]
> @holgerfriedrich, @lolodomo just FYI, it seems that #14970 caused #15019 to fail?

Sorry, apparently I'm mistaking mqtt.ruuvigateway and mqtt.espmilighthub.

**Comment by J-N-K** [comment]
@holgerfriedrich I just pushed a small addition to your branch. Is that ok with you?

**Comment by holgerfriedrich** [comment]
@J-N-K thanks! I could not make it work with the CI, now the test seems fine.

File: itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java [hunk 1]
```diff
@@ -132,6 +143,28 @@ public class RuuviGatewayTest extends MqttOSGiTest {
...
+            Class clazz = i.getArgument(0);
+            if (Temperature.class.equals(clazz)) {
+                return SIUnits.CELSIUS;
+            } else if (Acceleration.class.equals(clazz)) {
+                return Units.METRE_PER_SQUARE_SECOND;
+            } else if (Dimensionless.class.equals(clazz)) {
+                return Units.ONE;
+            } else if (ElectricPotential.class.equals(clazz)) {
+                return Units.VOLT;
+            } else if(Pressure.class.equals(clazz)) {
```
Comment (jlaur) [inline itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java:158]: Is Spotless disabled for tests? Just curious. I would have expected:
```suggestion
            } else if (Pressure.class.equals(clazz)) {
```

File: itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java [hunk 2]
```diff
@@ -132,6 +143,28 @@ public class RuuviGatewayTest extends MqttOSGiTest {
...
+                return HECTO(SIUnits.PASCAL);
+            } else if (Power.class.equals(clazz)) {
+                return Units.WATT;
+            }
+            return null;
+                }
```
Comment (jlaur) [inline itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java:164]: Indentation looks strange here, I guess that answers my previous question.

Comment (J-N-K) [reply itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java:164]: spotless is not disabled, it just seems to be not working for itests. It also doesn't work for all classes in core.

Comment (wborn) [reply itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/RuuviGatewayTest.java:164]: This is https://github.com/openhab/openhab-core/issues/3515

**Review by wborn — APPROVED** [review]
Thanks all!

**Review by J-N-K — APPROVED** [review]

**Pull request merged**